datadog/table_datadog_host.go

```diff
@@ -0,0 +1,101 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+
+	datadog "github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+	"github.com/turbot/steampipe-plugin-sdk/v3/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin/transform"
+)
+
+func tableDatadogHost(ctx context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "datadog_host",
+		Description: "A host is any piece of infrastructure that runs an instance of the Datadog Agent such as a bare metal instance or a VM.",
+		// There is no Datadog hosts endpoint for getting a single host so instead we use a filter to perform an exact match
+		Get: &plugin.GetConfig{
+			Hydrate:    listHosts,
+			KeyColumns: plugin.SingleColumn("name"),
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listHosts,
+			KeyColumns: plugin.KeyColumnSlice{
+				{Name: "name", Require: plugin.Optional},
```

Comment:
Can there be multiple hosts with the same name in a single account (or whatever the scope of a host is), or are they guaranteed to be unique?

Reply:
@cbruno10 The hostname parameter in my opinion is guaranteed to be unique. For example, when I install a datadog agent in an AWS instance, the host name is equivalent to the instance id. I don't seem to find any document which guarantees the uniqueness of a host.